src/main/java/uk/gov/hmcts/divorce/divorcecase/model/Solicitor.java

```diff
@@ -94,4 +104,52 @@ public boolean hasOrgId() {
         }
         return false;
     }
+
+    public String getAddress() {
+        if ((YesOrNo.NO == getAddressOverseas() || null == getAddressOverseas()) && (this.address != null && !this.address.equals(""))) {
+            String ukPostcodeRegex = "([A-Za-z][A-Ha-hJ-Yj-y]?[0-9][A-Za-z0-9]? ?[0-9][A-Za-z]{2}|[Gg][Ii][Rr] ?0[Aa]{2})";
+            Matcher matcher = Pattern.compile(ukPostcodeRegex).matcher(this.address);
+            if (matcher.find()) {
+                String postcode =  matcher.group(1);
+                String addressPostcodeStripped = this.address.replace(postcode, "")
+                    .replace("\n\n", "\n")
+                    .replace(",,", ",")
+                    .replace("\n,\n", "\n")
+                    .replace(", , ", ", ")
+                    .replace(" \n, \n", " \n")
+                    .replace("\n ,\n", "\n")
+                    .replace(", \n ,", ",")
+                    .replace(" \n  \n", " \n")
+                    .replace("\n \n ", "\n ")
+                    .replace(" \n \n", " \n")
+                    .stripTrailing();
+
+                if (addressPostcodeStripped.contains("\n")) {
+                    String upToFirstNewLine = addressPostcodeStripped.substring(0, addressPostcodeStripped.indexOf("\n"));
+                    if (upToFirstNewLine.endsWith(",") && !addressPostcodeStripped.endsWith(",")) {
+                        addressPostcodeStripped += ",";
+                    } else if (upToFirstNewLine.endsWith(", ")
+                        && (!addressPostcodeStripped.endsWith(",") && !addressPostcodeStripped.endsWith(", "))) {
+                        addressPostcodeStripped += ", ";
+                    } else if (upToFirstNewLine.endsWith(" ") && !addressPostcodeStripped.endsWith(" ")) {
+                        addressPostcodeStripped += " ";
+                    }
+                    addressPostcodeStripped += "\n";
+                    if (addressPostcodeStripped.substring(0, upToFirstNewLine.length() + 2).endsWith(" ")) {
+                        addressPostcodeStripped += " ";
+                    }
```

Comment:
Maybe we could use a regex for this too 🤔. We could check for a match against [, ] at the end of `upToFirstNewLine` and then append to the postcode stripped address if it isn't there already